### ISSUE TITLE
Error 'NoneType' object has no attribute 'id' in celery beat

### ISSUE THREAD

**Issue by Amitdedhia6** [celery/django-celery-beat#658]
### Summary:

Getting error 'NoneType' object has no attribute 'id' in celery beat

* Celery Version: 5.3.1
* Celery-Beat Version: 2.5.0

### Exact steps to reproduce the issue:
1. Refer detailed information section

### Detailed information

I am working on a Django based application (my django knowledge is limited, please let me know if I missed any steps or information). I am working with following modules.
Django==4.0.2
celery==5.3.1
django-celery-beat==2.5.0

I am working on Windows 10 Pro machine, on WSL2 with Docker Desktop.

In my application, I use both celery tasks as well as celery beat i.e. some tasks are executed as periodic tasks and some are executed on request basis. The deployment environment consists of 3 docker containers (1) Python container for main application (2) Container for celery (3) container for celery beat (there are more containers for migration and db etc, but not relevant in this context)

Containers for celery and celery beat have workers running by executing following commands:
Celery container command: `celery -A XXXXXX worker --loglevel=INFO`
Celery Beat container command: `celery -A XXXXXX  beat --loglevel=INFO`

`django_celery_beat` and  `django_celery_results` apps are installed (in all three containers) and migrations are run successfully.

Celery beat configuration is defined as follows:
```
CELERY_BEAT_SCHEDULE = {
      'check_provider_status': {
        'task': 'check_provider_status_task',
        'schedule': 60.0,
        'options': {
            'expires': 45.0,
        },
    },
}

```

These are additional settings I use:
```
CELERY_RESULT_BACKEND = 'django-db'
CELERY_ACCEPT_CONTENT = ['application/json']
CELERY_RESULT_SERIALIZER = 'json'
CELERY_TASK_SERIALIZER = 'json'
CELERY_TASK_IGNORE_RESULT = True
```
When the application starts, the celery beat schedules the task, the task runs (as per the logs in the celery container). However, as soon as the task returns, the celery beat crashes with following error.

```
2023-07-05 16:39:12 {"@timestamp":"2023-07-05T11:09:12.359Z","log.level":"info","message":"beat: Starting...","app":"web-django","ecs":{"version":"1.6.0"},"log":{"logger":"celery.beat","origin":{"file":{"line":629,"name":"beat.py"},"function":"start"},"original":"beat: Starting..."},"process":{"name":"MainProcess","pid":1,"thread":{"id":140535719381888,"name":"MainThread"}},"request_id":"none"}
2023-07-05 16:39:15 {"@timestamp":"2023-07-05T11:09:15.040Z","log.level":"info","message":"Scheduler: Sending due task check_provider_status (check_provider_status_task)","app":"web-django","ecs":{"version":"1.6.0"},"log":{"logger":"celery.beat","origin":{"file":{"line":278,"name":"beat.py"},"function":"apply_entry"},"original":"Scheduler: Sending due task check_provider_status (check_provider_status_task)"},"process":{"name":"MainProcess","pid":1,"thread":{"id":140535719381888,"name":"MainThread"}},"request_id":"none"}
2023-07-05 16:39:15 {"@timestamp":"2023-07-05T11:09:15.282Z","log.level":"critical","message":"beat raised exception <class 'AttributeError'>: AttributeError(\"'NoneType' object has no attribute 'id'\")","app":"web-django","ecs":{"version":"1.6.0"},"error":{"message":"'NoneType' object has no attribute 'id'","stack_trace":"  File \"/usr/local/lib/python3.9/site-packages/celery/apps/beat.py\", line 113, in start_scheduler\n    service.start()\n  File \"/usr/local/lib/python3.9/site-packages/celery/beat.py\", line 640, in start\n    interval = self.scheduler.tick()\n  File \"/usr/local/lib/python3.9/site-packages/celery/beat.py\", line 350, in tick\n    self.apply_entry(entry, producer=self.producer)\n  File \"/usr/local/lib/python3.9/site-packages/celery/beat.py\", line 285, in apply_entry\n    debug('%s sent. id->%s', entry.task, result.id)\n","type":"AttributeError"},"log":{"logger":"celery.beat","origin":{"file":{"line":115,"name":"beat.py"},"function":"start_scheduler"},"original":"beat raised exception <class 'AttributeError'>: AttributeError(\"'NoneType' object has no attribute 'id'\")"},"process":{"name":"MainProcess","pid":1,"thread":{"id":140535719381888,"name":"MainThread"}},"request_id":"none"}
2023-07-05 16:39:15 Traceback (most recent call last):
2023-07-05 16:39:15   File "/usr/local/bin/celery", line 8, in <module>
2023-07-05 16:39:15     sys.exit(main())
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/celery/__main__.py", line 15, in main
2023-07-05 16:39:15     sys.exit(_main())
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/celery/bin/celery.py", line 235, in main
2023-07-05 16:39:15     return celery(auto_envvar_prefix="CELERY")
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
2023-07-05 16:39:15     return self.main(*args, **kwargs)
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1055, in main
2023-07-05 16:39:15     rv = self.invoke(ctx)
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
2023-07-05 16:39:15     return _process_result(sub_ctx.command.invoke(sub_ctx))
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
2023-07-05 16:39:15     return ctx.invoke(self.callback, **ctx.params)
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/click/core.py", line 760, in invoke
2023-07-05 16:39:15     return __callback(*args, **kwargs)
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
2023-07-05 16:39:15     return f(get_current_context(), *args, **kwargs)
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/celery/bin/base.py", line 134, in caller
2023-07-05 16:39:15     return f(ctx, *args, **kwargs)
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/celery/bin/beat.py", line 72, in beat
2023-07-05 16:39:15     return beat().run()
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/celery/apps/beat.py", line 84, in run
2023-07-05 16:39:15     self.start_scheduler()
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/celery/apps/beat.py", line 113, in start_scheduler
2023-07-05 16:39:15     service.start()
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/celery/beat.py", line 640, in start
2023-07-05 16:39:15     interval = self.scheduler.tick()
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/celery/beat.py", line 350, in tick
2023-07-05 16:39:15     self.apply_entry(entry, producer=self.producer)
2023-07-05 16:39:15   File "/usr/local/lib/python3.9/site-packages/celery/beat.py", line 285, in apply_entry
2023-07-05 16:39:15     debug('%s sent. id->%s', entry.task, result.id)
2023-07-05 16:39:15 AttributeError: 'NoneType' object has no attribute 'id'

```
In `beat.py`, we have following code
```
    def apply_entry(self, entry, producer=None):
        info('Scheduler: Sending due task %s (%s)', entry.name, entry.task)
        try:
            result = self.apply_async(entry, producer=producer, advance=False)
        except Exception as exc:  # pylint: disable=broad-except
            error('Message Error: %s\n%s',
                  exc, traceback.format_stack(), exc_info=True)
        else:
            debug('%s sent. id->%s', entry.task, result.id)

```
The `result` object is `None`, and hence the `debug` statement crashes.

Why this could happen? Am I missing anything here? I am not setting up beat schedule from django admin ui. Is it required or must?

**Comment by Amitdedhia6** [comment]
I tried setting up the command from django admin UI instead of manual settings, got the same behaviour and results.

**Comment by sayan-bose** [comment]
If you are using a custom `Task`, and you and have modified the `apply_async` method, make sure that you are returning the result. Celery beat expects the `apply_async` method to return the result of the task.

something like this
```
from celery import Task

class CustomTask(Task):

    def apply_async(self, ...):
        return super().apply_async(...)

    ....
```

**Comment by sayan-bose** [comment]
Its not an issue with `django-celery-beat`, its just the way `celery` `beat`  works. You can refer to the `def apply_entry(): ...` method in `beat.py`  [link](https://github.com/celery/celery/blob/main/celery/beat.py).

**Comment by cclauss** [comment]
Closing.  This was fixed in https://github.com/celery/celery/pull/8428/files which was merged in ` v5.3.4` so please upgrade to a [current release of Celery](https://github.com/celery/celery/releases).
